frontend/packages/kubevirt-plugin/src/k8s/requests/vm/create/initialize-vm.ts

```diff
@@ -14,88 +14,83 @@ import { NetworkWrapper } from '../../../wrapper/vm/network-wrapper';
 import { NetworkInterfaceWrapper } from '../../../wrapper/vm/network-interface-wrapper';
 import { MutableVMWrapper } from '../../../wrapper/vm/vm-wrapper';
 import { getVolumeCloudInitNoCloud } from '../../../../selectors/vm';
-import { VolumeWrapper } from '../../../wrapper/vm/volume-wrapper';
+import { MutableVolumeWrapper, VolumeWrapper } from '../../../wrapper/vm/volume-wrapper';
 import {
   CloudInitDataFormKeys,
   CloudInitDataHelper,
 } from '../../../wrapper/vm/cloud-init-data-helper';
-import { DataVolumeWrapper } from '../../../wrapper/vm/data-volume-wrapper';
+import {
+  DataVolumeWrapper,
+  MutableDataVolumeWrapper,
+} from '../../../wrapper/vm/data-volume-wrapper';
 import { StorageUISource } from '../../../../components/modals/disk-modal/storage-ui-source';
 import { insertName, joinIDs } from '../../../../utils';
 import {
-  getDefaultSCAccessMode,
+  getDefaultSCAccessModes,
   getDefaultSCVolumeMode,
 } from '../../../../selectors/config-map/sc-defaults';
 import { VM_TEMPLATE_NAME_PARAMETER } from '../../../../constants/vm-templates';
 import { CreateVMEnhancedParams } from './types';
 
+const resolveDataVolumeName = (
+  dataVolumeWrapper: DataVolumeWrapper,
+  {
+    newName,
+    isTemplate,
+    isPlainDataVolume,
+  }: { newName: string; isTemplate: boolean; isPlainDataVolume: boolean },
```

Comment:
Just wondering, why do we need to wrap these arguments in an object?

Reply:
IMO, it is easier to navigate the code like this (eg https://github.com/openshift/console/pull/4217/files#diff-1a53f6615b700761ee940794dc53a83eR72)
first argument is base for the name and the second are the options which affect the name resolution